syft/execution/placeholder.py

```diff
@@ -97,14 +102,7 @@ def detail(worker: AbstractWorker, tensor_tuple: tuple) -> "PlaceHolder":
         tags = syft.serde.msgpack.serde._detail(worker, tags)
         description = syft.serde.msgpack.serde._detail(worker, description)
 
-        if not hasattr(worker, "_tmp_placeholders"):
-            worker._tmp_placeholders = {}
-
-        if tensor_id not in worker._tmp_placeholders:
-            tensor = PlaceHolder(owner=worker, id=tensor_id, tags=tags, description=description)
-            worker._tmp_placeholders[tensor_id] = tensor
-
-        return worker._tmp_placeholders[tensor_id]
+        return PlaceHolder(owner=worker, id=tensor_id, tags=tags, description=description)
```

Comment:
How do you make sure you don't create 2 PlaceHolder with the same tags?

Reply:
I don't, but I don't think it was done before neither

Reply:
The check to avoid duplicate placeholders moves after the deserialization. Since `Actions` no longer directly contain `Placeholders` (which move to a field on `Plan`), we don't have to worry about deserializing the same `Placeholder` multiple times as long as the property on `Plan` doesn't contain duplicates.